app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITemplatesTestController.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.debttransformationstub.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.Environment
+import play.api.libs.json.Json
+import play.api.mvc.{Action, ControllerComponents}
+import uk.gov.hmrc.debttransformationstub.config.AppConfig
+import uk.gov.hmrc.debttransformationstub.models.debtmanagment.templateRequest
+import uk.gov.hmrc.debttransformationstub.services.DebtManagementAPIPollingService
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.io.Source
+
+@Singleton()
+class DebtManagementAPITemplatesTestController @Inject()(
+  appConfig: AppConfig,
+  cc: ControllerComponents,
+  pollingService: DebtManagementAPIPollingService,
+  environment: Environment
+)(implicit val executionContext: ExecutionContext) extends BackendController(cc) {
+
+  private val basePath = "conf/resources/data"
+
+  def fieldCollectionsTemplates(): Action[templateRequest] =
+    Action.async(parse.tolerantJson[templateRequest]) { request =>
+      if (appConfig.isPollingEnv)
+        pollingService.insertRequestAndServeResponse(Json.toJson(request.body), request.uri).map {
+          case Some(response) => Status(response.status.getOrElse(200))(response.content)
+          case None => ServiceUnavailable
+        }
+      else
+        environment.getExistingFile(s"$basePath/dm.template/template.json") match {
```

Comment:
Consider renaming to indicate it is for fc...

Reply:
renamed file as suggested